core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -765,14 +807,16 @@ private[spark] class TaskSetManager(
       case e: TaskFailedReason =>  // TaskResultLost, TaskKilled, and others
         logWarning(failureReason)
         None
+    }
 
-      case e: TaskEndReason =>
-        logError("Unknown TaskEndReason: " + e)
-        None
+    // we might rack up a bunch of fetch-failures in rapid succession, due to a bad node.  But
```

Comment:
I might keep this comment in the FetchFailed task reason (easier to get out of date here), especially since the countTowardsTaskFailures variable is so clearly named